src/ruby/end2end/package_with_underscore_checker.rb

```diff
@@ -0,0 +1,54 @@
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'open3'
+require 'tmpdir'
+
+def main
+  root_dir = File.join(File.dirname(__FILE__), '..', '..', '..')
+  pb_dir = File.join(root_dir, 'src', 'ruby', 'end2end', 'protos')
+
+  fail 'CONFIG env variable unexpectedly unset' unless ENV['CONFIG']
+  bins_sub_dir = ENV['CONFIG']
+  bins_dir = File.join('.', '..', '..', '..', '..', 'bins', bins_sub_dir)
```

Comment:
thanks for doing this. I'm not sure if you can see the failed tests, but I think this bins_dir needs to be changed (looks about two directories off). the tests are failing right now because the binary file cant be found